### PR TITLE
ARP to IPA mapping, g2p_encode for IPATokenizer

### PR DESCRIPTION
**Pull request by ekmb**
Signed-off-by: ekmb <ebakhturina@nvidia.com>

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ X] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        arp_ipa_map_file: ARPABET to IPA phoneme mapping file
+        arp_dict_file: path to ARPABET version of CMU dictionary
+        output_ipa_file: path to output IPA version of CMU dictionary
+    """
+    arp_to_ipa_dict = _get_arpabet_to_ipa_mapping(arp_ipa_map_file)
+    with open(arp_dict_file, "r", encoding="utf-8") as f_arp, open(output_ipa_file, "w", encoding="utf-8") as f_ipa:
+        for line in f_arp:
+            if line.startswith(";;;"):
+                f_ipa.write(line)
+            else:
```
Comment (redoctopus) [inline scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py:127]: Does this handle entries with comments at the end of the line? We don't have any lines like this in our CMUdicts so it's probably fine if we don't to handle them, but I feel like I've seen them before from external ones.

File: scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        '--cmu_arpabet',
+        help="Path to CMU ARPABET dictionary file",
+        type=str,
+        default="tts_dataset_files/cmudict-0.7b_nv22.08",
+    )
+    parser.add_argument(
+        "--ipa_out",
+        help="Path to save IPA version of the dictionary",
+        type=str,
+        default="tts_dataset_files/ipa-cmudict-0.7b_nv22.08_new",
```
[the remark this answers is not in the document]
Comment (ekmb) [reply scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py]: ok, so I'm making this arg a required one and overwriting the existing IPA CMUdict with the updated version.

[the remark this answers is not in the document]
Comment (redoctopus) [reply scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py]: Thanks!

File: scripts/tts_dataset_files/cmudict-0.7b_nv22.08 [hunk 3]
```diff
@@ -55,6 +55,11 @@
 ;;;   - Merged with https://github.com/Alexir/CMUdict/blob/master/cmudict-0.7b
 ;;;   - British spellings omitted
 ;;;
+;;;   This dictionary was created using ARPABET to IPA phoneme mapping
+;;;   NeMo/scripts/tts_dataset_files/cmudict-arpabet_to_ipa_nv22.08.tsv
+;;;   and NeMo/scripts/dataset_processing/g2p/convert_cmu_arpabet_to_ipa.py
+;;;   conversion script
```
Comment (redoctopus) [inline scripts/tts_dataset_files/cmudict-0.7b_nv22.08:61]: Is this in the right file? This looks like it's in the ARPABET CMUdict rather than the IPA one.

Comment (ekmb) [reply scripts/tts_dataset_files/cmudict-0.7b_nv22.08]: indeed! good catch!

**Review by redoctopus — DISMISSED** [review]
LGTM!

**Review by redoctopus — APPROVED** [review]

**Pull request merged**